### PR TITLE
Removal of embedded azure driver

### PR DESCRIPTION
**Pull request by ssbarnea**
The same driver will be installable as a plugin named molecule-azure:
https://pypi.org/project/molecule-azure/

Partial: #2228
Signed-off-by: Sorin Sbarnea <ssbarnea@redhat.com>

### CODE REVIEW

File: molecule/api.py [hunk 1]
```diff
@@ -9,10 +13,13 @@
...
 
-    plugins = {
-        entry_point.name: entry_point.load()
-        for entry_point in pkg_resources.iter_entry_points('molecule_driver')
-    }
+    plugins = {}
+    for entry_point in pkg_resources.iter_entry_points('molecule_driver'):
+        try:
+            plugins[entry_point.name] = entry_point.load()
+        except Exception as e:
```
Comment (cloudnull) [inline molecule/api.py]: I think this might cause some unintended consequences. I think we should limit this exception to "KeyError".

Comment (ssbarnea) [reply molecule/api.py]: @cloudnull In fact this piece of code is also present in a separated review and I need it for lots of exceptions, what happens is that pkg_resources can throw a very wide range of exceptions, but redoing the plugin loading will be subject of another change.

Comment (cloudnull) [reply molecule/api.py]: understood. thanks for the explanation.

File: molecule/cookiecutter/molecule/{{cookiecutter.role_name}}/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/molecule.yml [hunk 2]
```diff
@@ -53,6 +51,8 @@ platforms:
 {%- elif cookiecutter.driver_name == 'vagrant' %}
   - name: instance
     box: ubuntu/xenial64
+{%- else %}
```
[the remark this answers is not in the document]
Comment (ssbarnea) [reply molecule/cookiecutter/molecule/{{cookiecutter.role_name}}/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/molecule.yml]: That is on purpose, so we do have an instance mentioned even when the driver nature is unknown to molecule, something that will become the rule when we move them out. I considered that is better to have an instance when mentioned in the template when the driver is another one. This also keeps the code backwards compatible with azure.

**Pull request merged**